test/parallel/test-dns-resolve-promises.js

```diff
@@ -0,0 +1,22 @@
+// Flags: --expose-internals
+'use strict';
+require('../common');
+const assert = require('assert');
+const { internalBinding } = require('internal/test/binding');
+const cares = internalBinding('cares_wrap');
+const EPERM = internalBinding('constants').os.errno.EPERM;
+const dnsPromises = require('dns').promises;
+
+// Stub cares to force an error so we can test DNS error code path.
+// Returning a negative value means the error corresponding to the postiive
+// value. (This is to distinguish system errors from Node.js errors.)
```

Comment:
I think this doesn’t distinguish between system errors/Node.js errors, but rather follows libuv by using negative values as error codes, which on POSIX match the negated `errno` values?

```suggestion
// value. (This is the convention libuv uses for reporting errors.)
```